File: app/src/main/java/com/my/televip/virtuals/messenger/LocaleController.java
package com.my.televip.virtuals.messenger;

import com.my.televip.Class.ClassNames;
import com.my.televip.Class.ClassLoad;
import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;

import java.util.Locale;

import de.robv.android.xposed.XposedHelpers;

public class LocaleController {

    Object localeController;

    public LocaleController(){
        localeController = XposedHelpers.callStaticMethod(ClassLoad.getClass(ClassNames.LOCALE_CONTROLLER), Obfuscate.getMethodName("LocaleController", "getInstance"));
    }

    public Locale getCurrentLocale() {
        return (Locale) XposedHelpers.getObjectField(localeController, Obfuscate.getFieldName("LocaleController", "currentLocale"));
    }

    public static boolean isRTL() {
        return (boolean) XposedHelpers.getStaticBooleanField(ClassLoad.getClass(ClassNames.LOCALE_CONTROLLER), Obfuscate.getFieldName("LocaleController", "isRTL"));
    }

}

File: app/src/main/java/com/my/televip/virtuals/messenger/LocaleController.kt
package com.my.televip.virtuals.messenger

import com.my.televip.Class.ClassLoad
import com.my.televip.Class.ClassNames
import com.my.televip.obfuscate.Obfuscate
import de.robv.android.xposed.XposedHelpers
import java.util.Locale

class LocaleController(@JvmField val localeController: Any? = null) {

    val currentLocale: Locale?
        get() {
            val target = localeController ?: getInstance()?.localeController ?: return null
            return XposedHelpers.getObjectField(target, Obfuscate.getFieldName("LocaleController", "currentLocale")) as? Locale
        }

    companion object {
        @JvmStatic
        fun getInstance(): LocaleController? {
            val lcClass = ClassLoad.getClass(ClassNames.LOCALE_CONTROLLER) ?: return null
            val instance = XposedHelpers.callStaticMethod(lcClass, Obfuscate.getMethodName("LocaleController", "getInstance"))
            return LocaleController(instance)
        }

        @JvmStatic
        fun isRTL(): Boolean {
            val lcClass = ClassLoad.getClass(ClassNames.LOCALE_CONTROLLER) ?: return false
            return XposedHelpers.callStaticMethod(lcClass, Obfuscate.getMethodName("LocaleController", "isRTL")) as? Boolean ?: false
        }
    }
}
